internal/pkg/vm/internal/result/result.go

```diff
@@ -0,0 +1,38 @@
+package result
+
+import (
+	"github.com/filecoin-project/go-filecoin/internal/pkg/encoding"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/internal/exitcode"
+)
+
+// InvocOutput is what is returned by an actor executing a method.
+type InvocOutput struct {
+	ExitCode    exitcode.ExitCode
+	ReturnValue []byte
+}
+
+// Success returns an empty succesfull result.
+func Success() InvocOutput {
+	return InvocOutput{ExitCode: 0}
+}
+
+// Value returns a successful code with the value encoded.
+//
+// Callers do NOT need to encode the value before calling this method.
+// Review: for @spec, @go-filecoin, changed this to an interface so the caller doesnt have to do the serialization himself
+func Value(obj interface{}) InvocOutput {
+	aux, err := encoding.Encode(obj)
```

Comment:
Since we're not going through the specialty abi encodings is this implicitly getting rid of the abi?  Or do all of those encodings simply wrap `encoding` anyway?